narwhals/_arrow/dataframe.py

```diff
@@ -90,10 +91,15 @@ def _change_version(self: Self, version: Version) -> Self:
             self._native_frame, backend_version=self._backend_version, version=version
         )
 
-    def _from_native_frame(self: Self, df: pa.Table) -> Self:
-        return self.__class__(
+    def _from_native_frame(
+        self: Self, df: pa.Table, *, validate_column_names: bool = False
```

Comment:
🤔 maybe the internal default should be true, and then we just opt out of the check when we know it's safe to do so